Framework/DataHandling/src/LoadNGEM.cpp

```diff
@@ -99,13 +100,14 @@ void addFrameToOutputWorkspace(
  * @param events The main events data.
  * @param dataWorkspace The workspace to add the data to.
  */
-void createEventWorkspace(const int &maxToF, const double &binWidth,
+void createEventWorkspace(const double &maxToF, const double &binWidth,
                           std::vector<DataObjects::EventList> &events,
                           DataObjects::EventWorkspace_sptr &dataWorkspace) {
   std::vector<double> xAxis;
   // Round up number of bins needed and reserve the space in the vector.
-  xAxis.reserve(int(std::ceil(maxToF / binWidth)));
-  for (auto i = 0; i < (maxToF / binWidth); i++) {
+  const int nbins{int(std::ceil(maxToF / binWidth))};
+  xAxis.reserve(nbins);
+  for (auto i = 0; i < nbins; i++) {
```

Comment:
This could just be `std::iota`.

(unrelated to the actual fix, but could just be done now)

Reply:
Ah yes I always forget about that algorithm as I find the name confusing. I'll swap it out.

Reply:
Will `iota` work here actually? It only increments the value using `++`. std::generate will do the job though...

Reply:
Ah, no it wouldn't. `std::generate` is the correct alternative.